core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -654,9 +654,27 @@ public void destroyCache(final String alias) throws CachePersistenceException {
     if (alias == null) {
       throw new NullPointerException("Alias cannot be null");
     }
+
     LOGGER.debug("Destroying Cache '{}' in {}.", alias, simpleName);
-    removeAndCloseWithoutNotice(alias);
-    destroyPersistenceSpace(alias);
+
+    StatusTransitioner.Transition maintenance = null;
+    try {
+      maintenance = statusTransitioner.maintenance();
+      maintenance.succeeded();
+    } catch(IllegalStateException e) {
+      // the cache manager is already started, no need to put it in maintenance
```

Comment:
Correct - it just makes sure that if we failed to put in maintenance because it is already in maintenance in another thread we do not blindly proceed. Makes sense?